tests/test_collector_sink.py

```diff
@@ -0,0 +1,109 @@
+# BSD 2-Clause License
+#
+# Copyright (c) 2021-2023, Hewlett Packard Enterprise
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# 1. Redistributions of source code must retain the above copyright notice, this
+#    list of conditions and the following disclaimer.
+#
+# 2. Redistributions in binary form must reproduce the above copyright notice,
+#    this list of conditions and the following disclaimer in the documentation
+#    and/or other materials provided with the distribution.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+# DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+# FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+# SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+# OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+import pathlib
+import typing as t
+import uuid
+
+import pytest
+
+from conftest import MockCollectorEntityFunc
+from smartsim._core.entrypoints.telemetrymonitor import FileSink
+
+# The tests in this file belong to the group_a group
+pytestmark = pytest.mark.group_a
+
+
+@pytest.mark.asyncio
+async def test_sink_null_filename(mock_entity: MockCollectorEntityFunc) -> None:
+    """Ensure the filesink handles a null filename as expected"""
+    with pytest.raises(ValueError):
+        # pass null file path
+        sink = FileSink(None)  # type: ignore
+
+
+@pytest.mark.asyncio
```

Comment:
The FileSink tests should monkeypatch the FileSink to write to a file-like object instead of the actual path. That way we don't have to worry about the filesystem actually making the file and dumping buffers in time leading to false failures

Reply:
Most tests are using a `MockSink` to meet this need. This test in particular is of the Sink implementation and cannot be mocked